extension/extension.go

```diff
@@ -45,6 +45,8 @@ type PipelineWatcher interface {
 // wishes to be notified of the Collector's effective configuration.
 type ConfigWatcher interface {
 	// NotifyConfig notifies the extension of the Collector's current effective configuration.
+	// The extension owns the `confmap.Conf`. Callers should ensure that it's safe to
+	// store the pointer `conf` and use it later concurrently
```

Comment:
```suggestion
	// The extension owns the `confmap.Conf`. Callers must ensure that it's safe for
	// extensions to store the `conf` pointer and use it concurrently with any other
	// instances of `conf`.
```
I think we should be explicit as possible here to make the flow clear. Feel free to reword this if you think it could be improved.